v2/internal/testcommon/samples_tester.go

```diff
@@ -0,0 +1,226 @@
+	if sample.Owner().Kind == resolver.ResourceGroupKind {
+		handleParentObject(sample, rg, samples.SamplesMap)
+	} else {
+		samples.childObjects[sample.GetObjectKind().GroupVersionKind().String()] = sample
+	}
+}
+
+func handleChildObjects(tc *KubePerTestContext, samples *SampleObject) {
+	for gvk, sample := range samples.childObjects {
+
+		specField := reflect.ValueOf(sample.GetSpec())
+		originalVersion := specField.MethodByName("OriginalVersion").Call([]reflect.Value{})
+
+		tc.Expect(originalVersion[0]).ToNot(BeNil())
+		tc.Expect(originalVersion).ToNot(HaveLen(0))
+		tc.Expect(originalVersion[0].String()).ToNot(BeEmpty())
+
+		version := originalVersion[0].String()
+		tree, ok := samples.SamplesMap[version]
+		// Parent version should always exist
+		tc.Expect(ok).To(BeTrue())
+
+		// get the parent object
+		val, ok := tree.Get(sample.Owner().Kind)
+		// If owner already does not exist, we check if its unvisited in 'childObjects' map.
+		// If it does, we wait until we visit the owner and add is to the tree or the test condition fails
+		if !ok {
+			ownerGVK := constructGVK(sample.Owner().Group, version, sample.Owner().Kind)
+			_, ok = samples.childObjects[ownerGVK]
+			tc.Expect(ok).To(BeTrue())
+			continue
+		}
+
+		owner := val.(genruntime.ARMMetaObject)
+		sample = setOwnersName(sample, owner.GetName())
+		tree.Put(sample.GetObjectKind().GroupVersionKind().Kind, sample)
+
+		// Delete after processing
+		delete(samples.childObjects, gvk)
+	}
+
+	// We need to do recursion here, as per the logic in this block, we check if we've visited the owner
+	// for current resource or not. If not, then we keep trying until we visit all the members of childObjects map.
+	if len(samples.childObjects) > 0 {
+		handleChildObjects(tc, samples)
+	}
+}
+
+func handleParentObject(sample genruntime.ARMMetaObject, rg *resources.ResourceGroup, samples map[string]*linkedhashmap.Map) {
+
+	version := sample.GetObjectKind().GroupVersionKind().GroupVersion().Version
+	sample = setOwnersName(sample, rg.Name)
+
+	kind := sample.GetObjectKind().GroupVersionKind().Kind
+	// TODO: If we have similar Kind in the same group and version, we need to put them in a new directory
+	if _, ok := samples[version]; ok {
+		samples[version].Put(kind, sample)
+	} else {
+		linkedHashMap := linkedhashmap.New()
+		linkedHashMap.Put(kind, sample)
+		samples[version] = linkedHashMap
+	}
+}
+
+func setOwnersName(sample genruntime.ARMMetaObject, ownerName string) genruntime.ARMMetaObject {
+	specField := reflect.ValueOf(sample.GetSpec()).Elem()
+	ownerField := specField.FieldByName("Owner").Elem()
+	ownerField.FieldByName("Name").SetString(ownerName)
+
+	return sample
+}
+
+func isExclusion(path string) bool {
+	for _, exclusion := range exclusions {
+		if strings.Contains(path, exclusion) {
+			return true
+		}
+	}
+	return false
+}
+
+func getGroupPath(group string) string {
+	return path.Join(samplesPath, group)
+}
+
+func constructGVK(group, version, kind string) string {
```

Comment:
Prefer to be explicit about the type of each parameter:

```suggestion
func constructGVK(group string, version string, kind string) string {
```